Add support for ms graph security v2 (#25772)
demisto/content: Packs/MicrosoftGraphSecurity/ReleaseNotes/2_1_19.md

#### Incident Fields

- New: **Graph Security Id**.
- New: **Microsoft Recommended Action**.
- New: **Microsoft Service Source**.
- New: **Microsoft Alert Determination**.

#### Incident Types

- New: **Graph Security Alert**.

#### Integrations
##### Microsoft Graph Security

- Updated the Docker image to: *demisto/crypto:1.0.0.54987*.

> - Added the command **msg-create-alert-comment**, use this command to create comments for the V2 alerts.

Author: The command is supported only by V2.
I'll change it to:
- Added the command **msg-create-alert-comment**, this command is supported only when using API V2.

#### Incident Fields

> 

Reviewer: ```suggestion

The following are new incident fields:

```

> - New: **Microsoft Graph Security Id**

Reviewer: ```suggestion
- **Microsoft Graph Security Id**
```

> - New: **Microsoft Graph Security Recommended Action**

Reviewer: ```suggestion
- **Microsoft Graph Security Recommended Action**
```

> - New: **Microsoft Graph Security Service Source**

Reviewer: ```suggestion
- **Microsoft Graph Security Service Source**
```

> - New: **Microsoft Graph Security Alert Determination**

Reviewer: ```suggestion
- **Microsoft Graph Security Alert Determination**
```

> - New: **Microsoft Graph Security Comment**

Reviewer: ```suggestion
- **Microsoft Graph Security Comment**
```

> - New: **Microsoft Graph Security Detector Id**

Reviewer: ```suggestion
- **Microsoft Graph Security Detector Id**
```

> - New: **Microsoft Graph Security Alert Evidence**

Reviewer: ```suggestion
- **Microsoft Graph Security Alert Evidence**
```

#### Incident Types

> - New: **Graph Security Alert**

Reviewer: ```suggestion
New: **Graph Security Alert**
```

#### Integrations
##### Microsoft Graph Security

- Updated the Docker image to: *demisto/crypto:1.0.0.55191*.

> - Added the command **msg-create-alert-comment**, this command is supported only when using API V2.

Reviewer: ```suggestion
- Added the command **msg-create-alert-comment**. This command is supported only when using API V2.
```

> - Added the **MS graph security version** integration param, this param will decide the version of the api to use. Note: the default version is V2 which causes breaking changes and demands reconfiguring with new credentials. If you're using self-deployed application please refer to the readme to the integration readme to the `Required Permissions -> API V2` section to learn about the required permissions. Otherwise, please refer to the link in the `Help` section in the integration configuration window.

Reviewer: ```suggestion
- Added the *MS graph security version* integration parameter. This parameter will determine the version of the API to use. Note: The default version is V2 which causes breaking changes and demands reconfiguring with new credentials. If you're using a self-deployed application, refer to the `Required Permissions -> API V2` section  in the integration readme to learn about the required permissions. Otherwise, refer to the link in the `Help` section in the integration configuration window.
```

> - Added the **Fetch incidents of the given service sources only.** integration param, use this param to filter fetched alerts by service source. **Only relevant when using V2 of the API**.

Reviewer: ```suggestion
- Added the *Fetch incidents of the given service sources only.* integration parameter. Use this param to filter fetched alerts by service source. **Only relevant when using V2 of the API**.
```

> - Added support for V2 of the API for the following commands: **msg-search-alerts**, **msg-get-alert-details**, **msg-update-alert**. For more information, please refer to the integration readme.

Reviewer: ```suggestion
- Added support for V2 of the API for the following commands:
   - ***msg-search-alerts***
   - ***msg-get-alert-details***
   - ***msg-update-alert*** 
   For more information, refer to the integration readme.
```

> - Note: changed the display name of the following integration params: **First fetch time range** and **How many incidents to fetch each time** to **First fetch timestamp (<number> <time unit>, e.g., 12 hours, 7 days)** and **Max incidents per fetch** respectively.

Reviewer: ```suggestion
- Changed the display name of the following integration parameters: 
   - *First fetch time range* to *First fetch timestamp (<number> <time unit>, e.g., 12 hours, 7 days)*
   - *How many incidents to fetch each time* to *Max incidents per fetch*
```

> - Added

Reviewer: ```suggestion

```

> - Added the **page**, **page_size** and **limit** arguments to **msg-search-alerts** command, please refer to the documentation for more information about the usage and limitations.

Reviewer: ```suggestion
- Added the following arguments to the ***msg-search-alerts*** command:
   - *page*
   - *page_size*
   -  *limit*
   Refer to the documentation for more information about the usage and limitations.
```

#### Mappers

##### New: Graph Security Alert

> - Added a mapper **Graph Security Alert**, use this mapper to map incidents fetched from the V2 of the api.

Reviewer: ```suggestion
Added the **Graph Security Alert** mapper. Use this mapper to map incidents fetched from the V2 version of the API.
```
